internal/locate/region_request.go

```diff
@@ -633,7 +645,7 @@ func (state *accessByKnownProxy) onSendFailure(bo *retry.Backoffer, selector *re
 }
 
 func (state *accessByKnownProxy) onNoLeader(selector *replicaSelector) {
-	selector.state = &invalidLeader{}
+	selector.state = &tryFollower{leaderIdx: state.leaderIdx, lastIdx: state.leaderIdx, fromAccessKnownLeader: true}
```

Comment:
This is for https://github.com/tikv/client-go/issues/1207

Since I need to test `replicaSelectorV2 ` has some beha behavior with old `replicaSelector`, so I just change the old `replicaSelector` behavior here. If I don't change this, I'll have to modify the `replicaSelectorV2` code to accommodate this strange behavior.